lib/logmatcher.c

```diff
@@ -435,6 +458,7 @@ log_matcher_pcre_re_feed_named_substrings(LogMatcher *s, LogMessage *msg, int *m
       /* Now we can scan the table and, for each entry, print the number, the name,
          and the substring itself.
        */
+      GString *formatted_name = scratch_buffers_alloc();
```

Comment:
this allocation can move inside the log_matcher_pcre_re_feed_value_by_name() function, so you don't have to pass it as an argument.